### DISCUSSION TITLE
how to redirect if account doesn't exist by provider?

### DISCUSSION THREAD

**Discussion by lawplatform** [nextauthjs/next-auth#9130] [Help] [answered]
I've created a signup function to store user information (name, email, password), and I want to integrate it with NextAuth. So, I want to check if a user logs in via the Google provider but hasn't signed up yet. In that case, I want to redirect them to the signup page.

However, I don't know how to implement this. I found that when using the signIn callback, it returns true or false.

```
async redirect({ url, baseUrl }) {
  // Allows relative callback URLs
  if (url.startsWith("/")) return `${baseUrl}${url}`;
  // Allows callback URLs on the same origin
  else if (new URL(url).origin === baseUrl) return url;
  return baseUrl;
},


async signIn(user, account, profile) {
  const emailExists = await checkCustomEmail(profile.email);
  if (emailExists) {
    console.log("Email exists. Confirming login.");
    return true; // Return false to prevent sign-in
  }
  console.log("Need to register. Redirecting to signup page.");
  router.push("/signup"); // Assuming you have access to the router
  return false;
}
```
try to add useRouter but it is fail because it is [next-auth]/route.ts
and callback url seemed to only succeffully login 

`<Button onClick={() => signIn('google', { callbackUrl: '/', redirect: false })}>`

**Comment by goerlitz** [comment]
have a look at the [docs](https://next-auth.js.org/configuration/callbacks#sign-in-callback): "Or you can return a URL to redirect to"

```js
async signIn(user, account, profile) {
  const emailExists = await checkCustomEmail(profile.email);
  if (emailExists) {
    return true; // Return false to prevent sign-in
  } else {
    return "/signup";
  }
}
```

**Reply by lawplatform** [reply]
so.. Is it impossible to add a variable to the return address? 
Like, return /signup?email=${profile.email}; I wanted to pass the email address as a parameter.

**Reply by goerlitz** [reply, marked as answer]
sure, you can do
```js
return '/signup?' + new URLSearchParams({ email: user.email || '' });
```

However, you may want to sanitize the email to combat vulnerabilities caused by invalid user input.

This is how I implemented it:
```js
import * as EmailValidator from 'email-validator';

const params =
  user.email && EmailValidator.validate(user.email)
    ? new URLSearchParams({ email: user.email })
    : '';
return '/signup?' + params;
```

**Comment by lawplatform** [comment]
thank ! it work out of box :)
